src/Task.ts

```diff
@@ -78,6 +81,39 @@ export function handleTimeLockExpirationSet(event: TimeLockExpirationSet): void
   task.save()
 }
 
+export function handleCustomVolumeLimitSet(event: CustomVolumeLimitSet): void {
+  const task = Task.load(event.address.toHexString())
+  if (task == null) return log.warning('Missing task entity {}', [event.address.toHexString()])
+
+  const customVolumeLimitId = getVolumeLimitId(task, event.params.token)
+
+  const customVolumeLimit = new CustomVolumeLimit(customVolumeLimitId)
+  customVolumeLimit.task = task.id
+  customVolumeLimit.token = loadOrCreateERC20(event.params.token).id
+  customVolumeLimit.limitToken = loadOrCreateERC20(event.params.limitToken).id
+  customVolumeLimit.amount = event.params.amount
+  customVolumeLimit.period = event.params.period
+
+  customVolumeLimit.save()
+  task.save()
+}
+
+export function handleDefaultVolumeLimitSet(event: DefaultVolumeLimitSet): void {
+  const task = Task.load(event.address.toHexString())
+  if (task == null) return log.warning('Missing task entity {}', [event.address.toHexString()])
+
+  const defaultVolumeLimitId = getVolumeLimitId(task, event.params.token)
+
```

Comment:
Since there is only one default volume limit per task, using the task address as the identifier should be fine. Otherwise note that every time you change the default volume limit you will be creating a new ID, therefore, a new entity

```suggestion
  const defaultVolumeLimitId = task.id
```